File: src/FirstTracks.Web/Scripts/Models/SkiResort/SkiResort.ts
import { Trail } from '../Trail/Trail'; 

export class SkiResort {
    skiResortId: string;
    name: string;
    city: string;
    state: string;
    imagePath: string;
    mountainHref: string;
    rating: number;
    tags: string[];
    trails: Trail[];

    constructor(
        skiResortId: string,
        name: string,
        city: string,
        state: string,
        imagePath: string,
        mountainHref: string,
        rating: number,
        tags: string[],
        trails: Trail[]
     ) {
        this.skiResortId = skiResortId;
        this.name = name;
        this.city = city;
        this.state = state;
        this.imagePath = imagePath;
        this.mountainHref = mountainHref;
        this.rating = rating;
        this.tags = tags;
        this.trails = trails;
    }
}

File: src/FirstTracks.Core/Models/SkiResort.cs
using System.Collections.Generic;

namespace FirstTracks.Core.Models
{
	public class SkiResort : BaseModel
	{
		public string SkiResortId { get; set; }
		public string Name { get; set; }
		public string ImagePath { get; set; }
		public List<Trail> Trails { get; set; }

		public SkiResort()
		{
			this.Trails = new List<Trail>();
		}
	}
}
